addons/account/models/chart_template.py

```diff
@@ -1,997 +1,735 @@
+                    for field_name in vals:
+                        field = self.env[model]._fields.get(field_name, None)
+                        if (field and
+                            field.relational and
+                            field.comodel_name not in created_models and
+                            (field.comodel_name in dict(all_data) or field.comodel_name == model)
+                        ):
+                            to_be_removed.append(field_name)
+                            to_delay[xml_id][field_name] = vals.get(field_name)
+                    for field_name in to_be_removed:
+                        del vals[field_name]
+                if any(to_delay.values()):
+                    all_data.append((model, to_delay))
+                yield model, data
+                created_models.add(model)
+
+        created_vals = {}
+        for model, data in defer(list(data.items())):
+            translate_vals = []
+            create_vals = []
+
+            for xml_id, record in data.items():
+                # Extract the translations from the values
+                if any('@' in key for key in record):
+                    translate_vals.append({
+                        translate.split('@')[1]: value
+                        for translate, value in record.items()
+                        if '@' in translate and value
+                    })
+                    translate_vals[-1]['en_US'] = record['name']
+                else:
+                    translate_vals.append(None)
+                for key in list(record):
+                    if '@' in key:
+                        del record[key]
+
+                # Manage ids given as database id or xml_id
+                if isinstance(xml_id, int):
+                    record['id'] = xml_id
+                    xml_id = False
+                else:
+                    xml_id = f"{('account.' + str(self.env.company.id) + '_') if '.' not in xml_id else ''}{xml_id}"
 
-    def generate_properties(self, acc_template_ref, company):
-        """
-        This method used for creating properties.
+                create_vals.append({
+                    'xml_id': xml_id,
+                    'values': deref(record, self.env[model]),
+                    'noupdate': True,
+                })
+            created_vals[model] = created = self.env[model]._load_records(create_vals)
+
+            # Update the translations in batch for all languages
+            translate_vals = [(r.id, Json(t)) for t, r in zip(translate_vals, created) if t]
+            if translate_vals:
+                self.env.cr.execute(f"""
+                    UPDATE "{self.env[model]._table}" AS m
+                    SET "name" =  t.value
+                    FROM (
+                        VALUES {', '.join(['(%s, %s::jsonb)'] * (len(translate_vals)))}
+                    ) AS t(id, value)
+                    WHERE m.id = t.id
+                """, [v for vals in translate_vals for v in vals])
```

Comment:
I think that using the `TranslationImporter` will be harder than anything else, I just wanted to avoid changing the ORM for something that is very specific for accounting.
The importer works by loading language by language whereas I already have all the translations available.